File: msg-entity/src/test/java/dev/number6/entity/dagger/TestChannelMessagesEntityComprehensionComponent.java
package dev.number6.entity.dagger;

import dev.number6.comprehend.results.PresentableEntityResults;
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction;
import dev.number6.message.ComprehensionResultsConsumer;
import dagger.Component;

import javax.inject.Singleton;

@Component(modules = {
        FakeCommonModule.class,
        ChannelMessagesHandlerModule.class,
        FakeComprehensionResultsModule.class
})
@Singleton
public interface TestChannelMessagesEntityComprehensionComponent extends ChannelMessagesEntityComprehensionComponent {

    @Singleton
    ChannelMessagesToComprehensionResultsFunction<PresentableEntityResults> getResultsFunction();

    @Singleton
    ComprehensionResultsConsumer<PresentableEntityResults> getConsumer();
}

File: msg-entity/src/test/kotlin/dev/number6/entity/dagger/TestChannelMessagesEntityComprehensionComponent.kt
package dev.number6.entity.dagger

import dagger.Component
import dev.number6.comprehend.results.PresentableEntityResults
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction
import dev.number6.message.ComprehensionResultsConsumer
import javax.inject.Singleton

@Component(modules = [FakeComprehendModule::class, ChannelMessagesHandlerModule::class, FakeComprehensionResultsModule::class])
@Singleton
interface TestChannelMessagesEntityComprehensionComponent : ChannelMessagesEntityComprehensionComponent {
    @Singleton
    fun getResultsFunction(): ChannelMessagesToComprehensionResultsFunction<PresentableEntityResults>

    @Singleton
    fun getConsumer(): ComprehensionResultsConsumer<PresentableEntityResults>
}